src/Types/ModelRelationsDynamicMethodReturnTypeExtension.php

```diff
@@ -103,35 +102,20 @@ public function getTypeFromMethodCall(
             return null;
         }
 
-        /** @var string $relatedModelClassName */
-        $relatedModelClassName = $this
-            ->relationParserHelper
-            ->findRelatedModelInRelationMethod($methodReflection);
-
-        if ((new ObjectType(BelongsTo::class))->isSuperTypeOf($returnType)->yes()) {
-            $classReflection = $methodReflection->getDeclaringClass();
-            $types           = [];
-
-            foreach (TypeUtils::flattenTypes($returnType) as $flattenType) {
-                if ((new ObjectType(BelongsTo::class))->isSuperTypeOf($flattenType)->yes()) {
-                    $types[] = $flattenType->getTemplateType(BelongsTo::class, 'TChildModel');
-                } else {
-                    $types[] = $flattenType->getTemplateType(Relation::class, 'TRelatedModel');
-                }
-            }
-
-            if (count($types) >= 2) {
-                $childType = new UnionType(array_map(static fn (Type $type) => new ObjectType($type->getObjectClassNames()[0]), $types));
-            } else {
-                $childType = new ObjectType($classReflection->getName());
-            }
-
-            return new GenericObjectType($returnTypeObjectClassNames[0], [
-                new ObjectType($relatedModelClassName),
-                $childType,
-            ]);
+        $models = $this->relationParserHelper->findModelsInRelationMethod($methodReflection);
+
+        $types   = array_map(static fn ($model) => new ObjectType((string) $model), $models);
+        $types[] = $scope->getType($methodCall->var);
+
+        if (
+            // @phpstan-ignore-next-line
```

Comment:
Again, this could be a bit more precise. Perhaps we can avoid this error altogether by declaring `LARAVEL_VERSION` a dynamic constant? See https://phpstan.org/config-reference#constants.

Reply:
Fixed!